src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapperTest.java

```diff
@@ -1,20 +1,266 @@
 
-import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestIntent;
+
+@SuppressWarnings("checkstyle:magicnumber")
+@ExtendWith(MockitoExtension.class)
 class ProcedureRequestMapperTest {
 
+    @InjectMocks
+    private ProcedureRequestMapper procedureRequestMapper;
+    @Mock
+    private UUIDGenerator uuidGenerator;
+    @Mock
+    private ResourceFullUrlGenerator fullUrlGenerator;
+
+    @Test
+    void testMapMessageToProcedureRequestNoPatientClinicalInfo() {
+        final Message message = new Message(Collections.emptyList());
+
+        final var procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null);
+
+        assertThat(procedureRequest).isEmpty();
+    }
+
+    @Test
+    void testMapToProcedureRequestThrowsExceptionWhenFreeTextIsMissing() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN"
+        ));
+
+        assertThatThrownBy(() -> procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null))
+            .isInstanceOf(FhirValidationException.class)
+            .hasMessage("Unable to map message. "
+                + "The FreeText segment is mandatory in Clinical Information");
+    }
+
+    @Test
+    void testMapToProcedureRequestThrowsExceptionWhenTypeOfClinicalObservationIsMissing() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "FTX+CID+++COELIAC"
+        ));
+
+        assertThatThrownBy(() -> procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null))
+            .isInstanceOf(MissingSegmentException.class)
+            .hasMessage("EDIFACT section is missing segment CIN");
+    }
+
+    @Test
+    void testMapToProcedureRequestFreeText() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+
+        final var procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null).get();
+
+        assertThat(procedureRequest.getNote()).hasSize(1)
+            .first()
+            .satisfies(note -> assertThat(note.getText()).isEqualTo("COELIAC"));
```

Comment:
Rather than us calling `Optional.get()` without an `isPresent()` check, we could re-write this slightly to assert that the `procedureRequest` `isNotEmpty`, before asserting it's contents:

```suggestion
                final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
            message, null, null, null, null, null);

        assertThat(procedureRequest).isNotEmpty();
        assertThat(procedureRequest.get().getNote())
            .hasSize(1)
            .first()
            .extracting(Annotation::getText)
            .isEqualTo("COELIAC");
```

This is similar to other tests where the mapper return type is Optional. See `PractitionerMapperTest.java`

Reply:
Sure! I have updated other tests in this class for consistency. Thanks!
8b7131fd2551c712915c350d8dd4f79d44ddf5d8